dali/test/python/test_fw_iterators.py

```diff
@@ -69,6 +71,61 @@ def create_pipeline(creator, batch_size, num_gpus):
     return pipes, iters
 
 
+def test_mxnet_iterator_model_fit():
+    from nvidia.dali.plugin.mxnet import DALIGenericIterator as MXNetIterator
+    import mxnet as mx
+    num_gpus = 1
+    batch_size = 1
+    class RN50Pipeline(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus, data_paths):
+            super(RN50Pipeline, self).__init__(batch_size, num_threads, device_id,)
+            self.input = ops.FileReader(file_root = data_paths, shard_id = device_id, num_shards = num_gpus)
+            self.decode_gpu = ops.ImageDecoder(device = "cpu", output_type = types.RGB)
+            self.res = ops.RandomResizedCrop(device="cpu", size =(224,224))
+
+            self.cmnp = ops.CropMirrorNormalize(device="cpu",
+                                                output_dtype=types.FLOAT,
+                                                output_layout=types.NCHW,
+                                                crop=(224, 224),
+                                                image_type=types.RGB,
+                                                mean=[0.485 * 255,0.456 * 255,0.406 * 255],
+                                                std=[0.229 * 255,0.224 * 255,0.225 * 255])
+            self.coin = ops.CoinFlip(probability=0.5)
+
+        def define_graph(self):
+            rng = self.coin()
+            jpegs, labels = self.input(name="Reader")
+            images = self.decode_gpu(jpegs)
+            images = self.res(images)
+            output = self.cmnp(images, mirror=rng)
+            return labels
+
+    pipes, _ = create_pipeline(lambda gpu: RN50Pipeline(batch_size=batch_size, num_threads=4, device_id=gpu, num_gpus=num_gpus,
+                                                                  data_paths=image_data_set), batch_size, num_gpus)
+    pipe = pipes[0]
+
+    class MXNetIteratorWrapper(MXNetIterator):
```

Comment:
Perhaps this could/should be moved to mxnet plugin and called `MXNetModelIterator` or similar? If anyone wants to use this API (which is obsolete, AFAIK?), they wouldn't have to copy-paste from test...

Reply:
Not really. It is made here more for this particular test than as a fully working feature. I think it won't work beyond single GPU and I don't know if we want to spend time figuring out how to make it compatible with such use case (as this approach is deprecated anyway).
The goal of this PR is to make MXNet iterator compatible with DataIter, current functions don't check that that why it has got out of our attention so far, but just in case it would be wise to have that done.